generated/src/aws-cpp-sdk-s3/source/S3ErrorMarshaller.cpp

```diff
@@ -127,27 +127,35 @@ AWSError<Aws::Client::CoreErrors> S3ErrorMarshaller::Marshall(const Aws::Http::H
   XmlDocument doc = XmlDocument::CreateFromXmlStream(body);
   body.seekg(readPointer);
   Aws::String bodyError;
+  Aws::String requestId;
 
   if (doc.WasParseSuccessful() &&
       !doc.GetRootElement().IsNull() && 
       doc.GetRootElement().GetName() == Aws::String("Error")) 
   {        
       //check if the first node fetched has no children
       auto messageNode = doc.GetRootElement().FirstChild("Message") ;
-      if(!messageNode.IsNull() && !messageNode.HasChildren())
+      if(!messageNode.IsNull())
       {
           message = messageNode.GetText();
       }
       auto codeNode = doc.GetRootElement().FirstChild("Code") ;
-      if(!codeNode.IsNull() && !codeNode.HasChildren())
+      if(!codeNode.IsNull())
       {
           bodyError = codeNode.GetText();
       }
+
+      auto requestIdNode = doc.GetRootElement().FirstChild("RequestId") ;
```

Comment:
is this more for logging?

Reply:
no it needs to be returned so that the user has the request ID so that they can provide it to AWS support for inquiring about issues